qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -45,14 +45,24 @@ class Initialize(Instruction):
     def __init__(self, params, num_qubits=None):
         """Create new initialize composite.
 
-        params (str or list):
+        params (str, list, or int):
           * list: vector of complex amplitudes to initialize to.
           * string: labels of basis states of the Pauli eigenstates Z, X, Y. See
                :meth:`~qiskit.quantum_info.states.statevector.Statevector.from_label`.
                Notice the order of the labels is reversed with respect to the qubit index to
                be applied to. Example label '01' initializes the qubit zero to `|1>` and the
-               qubit one to `|0>`
+               qubit one to `|0>`.
+          * int: an integer that is used as a bitmap indicating which qubits to initialize
+               to `|1>`. Example: setting params to 5 would initialize qubit 0 and qubit 2
+               to `|1>` and qubit 1 to `|0>`.
+        num_qubits (int): This parameter is only used if params is an int. Indicates the total
+            number of qubits in the `initialize` call. Example: `initialize` covers 5 qubits
+            and params is 3. This allows qubits 0 and 1 to be initialized to `|1>` and the
+            remaining 3 qubits to be initialized to `|0>`.
         """
+        if not isinstance(params, int) and num_qubits:
```

Comment:
Using `None` as boolean created some nasty bugs in the past (e.g. when the signature changed and suddenly ints or booleans were passed as positional args), it's just a little overhead to add `is not None` but much safer 🙂 
```suggestion
        if not isinstance(params, int) and num_qubits is not None:
```

Reply:
Agreed. Done in https://github.com/Qiskit/qiskit-terra/pull/5418/commits/e918604354719e4a05b696ba09a586f6a390a037